astroid/brain/brain_functools.py

```diff
@@ -103,6 +103,7 @@ class PartialFunction(astroid.FunctionDef):
 
         filled_positionals = len(call.positional_arguments[1:])
         filled_keywords = list(call.keyword_arguments)
+        is_partial_function = True
```

Comment:
@hippo91 I wonder if we can extract `PartialFunction` somehow from this module. We can use `astroid.objects` for it, and if we need to figure out if a function is partial or not, we can implement the `name` property and `qname` function, just like we do for `Super`, `FrozenSet` and friends. And instead of checking if `is_partial_function` is given, we only look for `.name` against a predefined list of elements for which to apply the condition in the mixin. Do you think that's worth it?

Reply:
@PCManticore it looks interesting.
I work on it.